File: backend-java/SpringCloudPXL/shoppingcart-service/src/main/java/be/pxl/services/domain/ShoppingCartProduct.java
package be.pxl.services.domain;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "shoppingcartproduct")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ShoppingCartProduct {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private Long productId;
    @Transient
    private Product product;
    private int quantity;
    @ManyToOne
    @JoinColumn(name="shoppingcart_id")
    private ShoppingCart shoppingCart;
}

File: frontend-web/e-commerceplatform/src/app/models/shoppingcart-product.ts
import { Product } from "./product";

export class ShoppingCartProduct {
  id?: number;
  productId: number;
  product: Product;

  constructor(productId: number, product: Product, id?: number) {
    this.id = id;
    this.productId = productId;
    this.product = product;
  }
}